### PR TITLE
Add filtering support for duplicate files in nested projects

### LINKED ISSUE #143: Add filter to File Search to remove nested project duplicate files
When nested projects are searched, multiple matches will occur for each project that is above the matched file.  The file should be opened in the context of the innermost project and this is already done, but some users might want to pare down the list of matches to exclude the duplicates that don't occur in the innermost project.  This is an enhancement request to add a filter to do this in the Search result view.

### PR DESCRIPTION
**Pull request by jjohnstn**
- add filter to File Search to remove duplicate file entries that are attributed to different projects in a nested project scenario
- fixes #143

### CODE REVIEW

**Comment by jjohnstn** [comment]
Hi @mickaelistria please review.  This is based on @merks suggestion to avoid duplicates via a filter rather than in the search resuilts.

File: org.eclipse.search/search/org/eclipse/search/internal/ui/SearchMessages.properties [hunk 1]
```diff
@@ -97,6 +97,9 @@ SearchPage_wholeWord= Who&le word
 
 TextSearchEngine_statusMessage= Problems encountered during text search.
 
+TextSearchInnermostProjectFilter_action_label= &Nested project duplicates
```
Comment (mickaelistria) [inline org.eclipse.search/search/org/eclipse/search/internal/ui/SearchMessages.properties:100]:
```suggestion
TextSearchInnermostProjectFilter_action_label=Show only most &nested match
```
?

File: org.eclipse.search/search/org/eclipse/search/internal/ui/SearchMessages.properties [hunk 2]
```diff
@@ -97,6 +97,9 @@ SearchPage_wholeWord= Who&le word
...
+TextSearchInnermostProjectFilter_name= Nested project duplicates
```
Comment (mickaelistria) [inline org.eclipse.search/search/org/eclipse/search/internal/ui/SearchMessages.properties:101]:
```suggestion
TextSearchInnermostProjectFilter_name=Duplicate match from outer project
```
?

**Comment by jjohnstn** [comment]
Thanks @mickaelistria I used your string commits (interesting feature of github).  I don't think the filtered match label stuff needs to be sectioned out as it is essentially copied from the Java search page and just modified to take into account if the limit is also reached.  As well, there weren't any filters being set up for the File page before.  This plus the quick search change might be noteworthy additions, but I'll let you decide.

**Pull request merged**